assets/src/edit-story/components/library/karma/shapes/shapes.karma.js

```diff
@@ -86,3 +93,97 @@ describe('Shape library integration', () => {
+    await fixture.events.click(fixture.editor.library.shapesTab);
+    await fixture.events.click(
+      fixture.editor.library.shapes.sticker(testStickerType)
+    );
+
+    // Now background + 1 extra element
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(2);
+  });
+
+  it('add sticker via keyboard naviation to sticker preview', async () => {
+    // Only background initially
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(1);
+
+    // Switch to shapes tab and click the first sticker
+    await fixture.events.click(fixture.editor.library.shapesTab);
+    let limit = 0;
+    const stickersTab = fixture.editor.library.shapes.getByTestId(
+      'stickers-library-pane'
+    );
+    while (!stickersTab.contains(document.activeElement) && limit < 10) {
+      // eslint-disable-next-line no-await-in-loop
+      await fixture.events.keyboard.press('tab');
+      limit++;
+    }
+    await fixture.events.keyboard.press('Enter');
+
+    // Now background + 1 extra element
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(2);
+  });
+
+  it('add sticker via dragging from sticker preview', async () => {
+    // Only background initially
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(1);
+
+    // Switch to the shapes tab and drag the first sticker to the canvas
+    await fixture.events.click(fixture.editor.library.shapesTab);
+    const stickerButton =
+      fixture.editor.library.shapes.sticker(testStickerType);
+    const bgFrame = fixture.editor.canvas.framesLayer.frames[0].node;
+    await fixture.events.mouse.seq(({ moveRel, down, up }) => [
+      moveRel(stickerButton, 10, 10),
+      down(),
+      /* The steps give time for Moveable to react and display a clone to drag */
+      moveRel(bgFrame, 50, 50, { steps: 20 }),
+      up(),
+    ]);
+
+    // Now background + 1 extra element
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(2);
+  });
+
+  it('should not add sticker dragged out of the page area', async () => {
+    // Only background initially
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(1);
+
+    // Switch to the shapes tab and drag the first sticker to the canvas
+    await fixture.events.click(fixture.editor.library.shapesTab);
+    const stickerButton =
+      fixture.editor.library.shapes.sticker(testStickerType);
+    const bgFrame = fixture.editor.canvas.framesLayer.frames[0].node;
+
+    // Shape is 1/3 of the page's width by default.
+    const { width: pageWidth } = bgFrame.getBoundingClientRect();
+    await fixture.events.mouse.seq(({ moveRel, down, up }) => [
+      moveRel(stickerButton, 10, 10),
+      down(),
+      /* The steps give time for Moveable to react and display a clone to drag */
+      moveRel(bgFrame, -(pageWidth / 3 + 20), 50, { steps: 20 }),
+      up(),
+    ]);
+
+    // Still only background.
+    expect(fixture.editor.canvas.framesLayer.frames.length).toBe(1);
+  });
```

Comment:
nit: add one more check? Something like this. It's a habit we're trying to get into on new features.  
```
it('should not have any accessibility violations', async () => {
 await expectAsync(fixture.editor.library.shapes.getByTestId('stickers-library-pane')).toHaveNoViolations();
})
```